scripts/db_migrator.py

```diff
@@ -125,6 +126,50 @@ def migrate_interface_table(self):
                 self.configDB.set_entry(table, key[0], data[key])
                 if_db.append(key[0])
 
+    def migrate_mgmt_ports_on_s6100(self):
+        '''
+        During warm-reboot, add back two 10G management ports which got removed from 6100
+        to ensure no change in bcm.config from older image
+        '''
+        if device_info.is_warm_restart_enabled('swss') == False:
+            log.log_notice("Skip migration on {}, warm-reboot flag not set".format(self.hwsku))
+            return True
+        
+        portCount = self.appDB.get(self.appDB.APPL_DB, 'PORT_TABLE:PortConfigDone', 'count')
+        if portCount == '':
+            log.log_notice("Port count not found - {}, skip migration".format(portCount))
+            return True
+
+        self.portCount = int(portCount)
+        if self.portCount > 64:
+            log.log_notice("Port count is {}, no migration needed".format(self.portCount))
+            return True        
+        
+        log.log_notice("Migrating management ports Ethernet64 & Ethernet65 to configDB for warm-reboot on {}".format(self.hwsku))
+        self.configDB.set_entry('PORT', 'Ethernet64', {'alias': 'tenGige1/1', 'description': 'tenGige1/1', 'index': '64', 'lanes': '129', 'mtu': '9100', 'pfc_asym': 'off', 'speed': '10000'})
```

Comment:
Question: Is there any chance that Ethernet64 and/or Ethernet65 already exists on configDB already?  Also, are we assuming that these two ports MUST be the two mgmt ports?  The previous portCount check seems to suggest that as long as the port count is 64 or less then this will always be the case... Not sure if there are any other S6100 variants currently or maybe in the future that this is not the case?

Reply:
Updated code based on discussion. Please review